Song.Tests/SongWriterTests.cs

```diff
@@ -0,0 +1,132 @@
+using System.Text;
+
+namespace Song.Tests
+{
+    public class SongWriterTests
+    {
+        [Fact]
+        public void Sing_SingleAnimal()
```

Comment:
Your tests aren't unit tests, because you're testing both SongWriter and VerseWriter.


Look into mocking using Moq: https://github.com/devlooped/moq

Reply:
Implemented mocks for the SongWriter tests using Moq.